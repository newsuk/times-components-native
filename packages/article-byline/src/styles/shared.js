import styleguide, { spacing } from "@times-components-native/styleguide";

const { colours, fontFactory } = styleguide();
const shared = {
  link: {
    ...fontFactory({
      font: "supporting",
      fontSize: "cardMeta",
    }),
    color: colours.functional.action,
    textDecorationLine: "none",
  },
  nonLinkText: {
    ...fontFactory({
      font: "supporting",
      fontSize: "cardMeta",
    }),
    color: colours.functional.primary,
    flexDirection: "row",
  },
  opinion: {
    ...fontFactory({
      font: "headline",
      fontSize: "smallHeadline",
    }),
    color: colours.section.comment,
    fontWeight: "400",
  },
  centered: {
    textAlign: "center",
  },
  text: {
    ...fontFactory({
      font: "supporting",
      fontSize: "cardMeta",
    }),
    color: colours.functional.secondary,
    flexDirection: "row",
  },
  unhighlighted: {
    marginTop: -7,
    paddingTop: 7,
    paddingBottom: 6,
    marginHorizontal: spacing(2),
    marginBottom: spacing(1),
  },
  highlighted: {
    borderWidth: 2,
    borderColor: colours.functional.action,
    borderRadius: 2,
    paddingTop: 5,
    paddingBottom: 4,
  },
};

export default shared;
