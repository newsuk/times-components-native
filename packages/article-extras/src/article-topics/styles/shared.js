import { colours, tabletWidth, narrowArticleContentWidth } from "@times-components-native/styleguide";

const sharedStyles = {
  topicsContainer: {
    borderTopColor: colours.functional.keyline,
    borderTopWidth: 1
  },
  topicsContainerTablet: {
    alignSelf: "center",
    width: tabletWidth
  },
  narrow: {
    width: narrowArticleContentWidth
  },
  topicsMetaContainer: {
    justifyContent: "flex-start"
  }
};

export default sharedStyles;
