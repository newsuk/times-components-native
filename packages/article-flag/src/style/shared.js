import { spacing, fontFactory } from "@times-components-native/styleguide";

const shared = {
  bullet: {
    borderRadius: 2.5,
    height: 5,
    width: 5,
    marginBottom: 3,
  },
  flagPadding: {
    marginRight: spacing(3),
  },
  flagsContainer: {
    marginBottom: spacing(3),
    marginTop: spacing(1),
  },
  flags: {
    flexDirection: "row",
    flexWrap: "wrap",
    marginBottom: spacing(2),
  },
  title: {
    ...fontFactory({
      font: "body",
      fontSize: "cardMetaMobile",
    }),
    fontWeight: "bold",
    lineHeight: 10,
    textTransform: "uppercase",
    fontSize: 10,
    letterSpacing: 0,
    includeFontPadding: false,
    paddingBottom: spacing(0),
    marginLeft: spacing(1),
    marginBottom: spacing(0),
  },
  view: {
    alignItems: "center",
    flexDirection: "row",
    marginTop: 0,
  },
};

export default shared;
