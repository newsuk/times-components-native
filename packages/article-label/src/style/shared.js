import styleguide from "@times-components-native/styleguide";
import { StyleSheet } from "react-native";

const { fontFactory } = styleguide();
const styles = StyleSheet.create({
  title: {
    ...fontFactory({
      font: "supporting",
      fontSize: "cardMetaMobile",
    }),
    fontWeight: "400",
    lineHeight: 11,
    marginBottom: 0,
    marginTop: -1,
    paddingTop: 1,
  },
});

export default styles;
