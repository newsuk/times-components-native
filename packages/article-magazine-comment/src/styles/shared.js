import styleguide from "@times-components-native/styleguide";

const { colours, fontFactory, spacing } = styleguide();
const sharedStyles = {
  articleHeadline: {
    ...fontFactory({
      font: "headline",
      fontSize: "headline",
    }),
    color: colours.functional.brandColour,
    marginBottom: spacing(2),
    textAlign: "center",
  },
  authorImage: {
    marginBottom: spacing(5),
  },
  container: {
    alignItems: "center",
    marginBottom: spacing(5),
    marginLeft: spacing(2),
    marginRight: spacing(2),
    paddingTop: spacing(7),
  },
  containerWithMargin: {
    marginBottom: 100,
  },
  datePublication: {
    ...fontFactory({
      font: "supporting",
      fontSize: "cardMeta",
    }),
    color: colours.functional.secondary,
    marginTop: spacing(3),
  },
  label: {
    marginBottom: spacing(2),
  },
  leadAssetContainer: {
    marginBottom: spacing(4),
  },
  meta: {
    flexDirection: "row",
    flexWrap: "wrap",
  },
  metaContainer: {
    alignItems: "center",
  },
  separator: {
    borderRightColor: colours.functional.keyline,
    borderRightWidth: 1,
    height: spacing(3),
    marginHorizontal: spacing(2),
  },
  standFirst: {
    ...fontFactory({
      font: "headlineRegular",
      fontSize: "smallestHeadline",
    }),
    color: colours.functional.primary,
    marginBottom: spacing(3),
    paddingHorizontal: spacing(2),
    textAlign: "center",
  },
};

export default sharedStyles;
