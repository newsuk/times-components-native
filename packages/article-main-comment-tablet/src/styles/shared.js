import styleguide, { tabletWidth, narrowArticleContentWidth } from "@times-components-native/styleguide";

const { colours, fontFactory, spacing } = styleguide();
const sharedStyles = {
  articleHeadline: {
    ...fontFactory({
      font: "headline",
      fontSize: "articleHeadline"
    }),
    color: colours.functional.brandColour,
    marginBottom: spacing(2),
  },
  authorImage: {
    width: "100%",
  },
  datePublication: {
    ...fontFactory({
      font: "supporting",
      fontSize: "cardMeta"
    }),
    color: colours.functional.secondary,
    marginTop: "auto"
  },
  header: {
    marginBottom: spacing(3),
    paddingBottom: spacing(4),
    paddingTop: spacing(7),
    maxWidth: narrowArticleContentWidth,
  },
  container: {
    paddingLeft: spacing(2),
    paddingRight: spacing(2)
  },
  label: {
    marginBottom: spacing(2)
  },
  metaContainer: {
    flexDirection: "row",
    width: "100%",
    paddingVertical: spacing(2),
    borderBottomColor: colours.functional.keyline,
    borderBottomWidth: 1,
    borderTopColor: colours.functional.keyline,
    borderTopWidth: 1
  },
  standFirst: {
    ...fontFactory({
      font: "headlineRegular",
      fontSize: "smallestHeadline"
    }),
    color: colours.functional.primary,
    marginBottom: spacing(3),
  },
  leftColumnContainer: {
    width: "25%",    
    zIndex: 1111,
    borderRightWidth: 1,
    borderColor: colours.functional.keyline,
    height: "100%",
    paddingLeft: 40,
    position: "absolute",
  },
  authorImage: {
    width: 130,
    marginLeft: 30,
  },
  bylines: {
    paddingTop: spacing(1),
    width: "100%",
  }
};

export default sharedStyles;
