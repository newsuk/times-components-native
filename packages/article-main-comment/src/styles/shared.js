import styleguide from "@times-components-native/styleguide";

const { colours, fontFactory, spacing } = styleguide();
const sharedStyles = {
  articleHeadline: {
    ...fontFactory({
      font: "headline",
      fontSize: "headline",
    }),
    color: colours.functional.brandColour,
    marginBottom: spacing(2),
    textAlign: "center",
  },
  authorImage: {
    marginBottom: spacing(5),
  },
  container: {
    alignItems: "center",
    paddingLeft: spacing(2),
    paddingRight: spacing(2),
  },
  containerWithMargin: {
    marginBottom: 65,
  },
  datePublication: {
    ...fontFactory({
      font: "supporting",
      fontSize: "cardMeta",
    }),
    color: colours.functional.secondary,
    marginTop: spacing(1),
  },
  header: {
    marginBottom: spacing(3),
    paddingBottom: spacing(4),
    paddingTop: spacing(7),
  },
  label: {
    marginBottom: spacing(2),
  },
  meta: {
    flexDirection: "row",
    flexWrap: "wrap",
  },
  metaContainer: {
    alignItems: "center",
    width: "100%",
    paddingVertical: spacing(2),
    borderBottomColor: colours.functional.keyline,
    borderBottomWidth: 1,
    borderTopColor: colours.functional.keyline,
    borderTopWidth: 1,
    zIndex: 10,
  },
  separator: {
    borderRightColor: colours.functional.keyline,
    borderRightWidth: 1,
    height: spacing(3),
    marginHorizontal: spacing(2),
  },
  standFirst: {
    ...fontFactory({
      font: "headlineRegular",
      fontSize: "smallestHeadline",
    }),
    color: colours.functional.primary,
    marginBottom: spacing(3),
    paddingHorizontal: spacing(2),
    textAlign: "center",
  },
};

export default sharedStyles;
