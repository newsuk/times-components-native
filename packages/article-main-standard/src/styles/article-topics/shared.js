import { colours } from "@times-components-native/styleguide";

const sharedStyles = {
  topicsContainer: {
    borderTopColor: colours.functional.keyline,
    borderTopWidth: 1
  },
  topicsMetaContainer: {
    justifyContent: "flex-start"
  }
};

export default sharedStyles;
