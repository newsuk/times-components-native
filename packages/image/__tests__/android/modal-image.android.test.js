import shared from "../modal-shared";

shared();
