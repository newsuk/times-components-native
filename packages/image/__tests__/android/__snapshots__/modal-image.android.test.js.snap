// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1. modal image 1`] = `
<View>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    presentationStyle="fullScreen"
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    visible={false}
  >
    <View>
      <View>
        <View
          focusable={true}
        >
          <Image
            resizeMode="contain"
            source={
              Object {
                "testUri": "../../../packages/image/assets/close-button.png",
              }
            }
          />
        </View>
      </View>
      <ImageZoomView
        enablePreload={true}
        enableSwipeDown={true}
        imageUrls={
          Array [
            Object {
              "aspectRatio": 2,
              "caption": "Caption",
              "credits": "Credits",
              "imageIndex": 0,
              "relativeHeight": 1,
              "relativeHorizontalOffset": 1,
              "relativeVerticalOffset": 1,
              "relativeWidth": 1,
              "uri": "http://example.com/image.jpg?crop=1016%2C677%2C0%2C0",
              "url": "http://example.com/image.jpg?crop=1016%2C677%2C0%2C0&offline=true",
            },
          ]
        }
        index={0}
        saveToLocalByLongPress={false}
        useNativeDriver={true}
      />
      <View
        pointerEvents="none"
      >
        <View>
          <View>
            <Text
              testID={null}
            >
              Caption
            </Text>
            <Text
              testID={null}
            >
              CREDITS
            </Text>
          </View>
        </View>
      </View>
    </View>
  </Modal>
  <View
    focusable={true}
  >
    <View>
      <Image
        borderRadius={0}
        fadeDuration={0}
        source={
          Object {
            "testUri": "../../../packages/responsive-image/assets/t.png",
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`2. modal image with no caption 1`] = `
<View>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    presentationStyle="fullScreen"
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    visible={false}
  >
    <View>
      <View>
        <View
          focusable={true}
        >
          <Image
            resizeMode="contain"
            source={
              Object {
                "testUri": "../../../packages/image/assets/close-button.png",
              }
            }
          />
        </View>
      </View>
      <ImageZoomView
        enablePreload={true}
        enableSwipeDown={true}
        imageUrls={
          Array [
            Object {
              "aspectRatio": 2,
              "caption": undefined,
              "credits": undefined,
              "imageIndex": 0,
              "relativeHeight": 1,
              "relativeHorizontalOffset": 1,
              "relativeVerticalOffset": 1,
              "relativeWidth": 1,
              "uri": "http://example.com/image.jpg?crop=1016%2C677%2C0%2C0",
              "url": "http://example.com/image.jpg?crop=1016%2C677%2C0%2C0&offline=true",
            },
          ]
        }
        index={0}
        saveToLocalByLongPress={false}
        useNativeDriver={true}
      />
      <View
        pointerEvents="none"
      />
    </View>
  </Modal>
  <View
    focusable={true}
  >
    <View
      accessibilityIgnoresInvertColors={true}
    >
      <Image
        fadeDuration={0}
        source={
          Object {
            "testUri": "../../../packages/responsive-image/assets/t.png",
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`3. modal image with custom highressize 1`] = `
<View>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    presentationStyle="fullScreen"
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    visible={false}
  >
    <View>
      <View>
        <View
          focusable={true}
        >
          <Image
            resizeMode="contain"
            source={
              Object {
                "testUri": "../../../packages/image/assets/close-button.png",
              }
            }
          />
        </View>
      </View>
      <ImageZoomView
        enablePreload={true}
        enableSwipeDown={true}
        imageUrls={
          Array [
            Object {
              "aspectRatio": 2,
              "caption": "Caption",
              "credits": "Credits",
              "imageIndex": 0,
              "relativeHeight": 1,
              "relativeHorizontalOffset": 1,
              "relativeVerticalOffset": 1,
              "relativeWidth": 1,
              "uri": "http://example.com/image.jpg?crop=1016%2C677%2C0%2C0",
              "url": "http://example.com/image.jpg?crop=1016%2C677%2C0%2C0&offline=true",
            },
          ]
        }
        index={0}
        saveToLocalByLongPress={false}
        useNativeDriver={true}
      />
      <View
        pointerEvents="none"
      >
        <View>
          <View>
            <Text
              testID={null}
            >
              Caption
            </Text>
            <Text
              testID={null}
            >
              CREDITS
            </Text>
          </View>
        </View>
      </View>
    </View>
  </Modal>
  <View
    focusable={true}
  >
    <View>
      <Image
        borderRadius={0}
        fadeDuration={0}
        source={
          Object {
            "testUri": "../../../packages/responsive-image/assets/t.png",
          }
        }
      />
    </View>
  </View>
</View>
`;
